rdr_service/dao/participant_summary_dao.py

```diff
@@ -1015,6 +1015,19 @@ def add_to_sqlalchemy_query(self, query, field):
                     passive_criterion,
                     ParticipantSummary.ehrReceiptTime > eighteen_month_ago
                 )
+            elif self.value == 'UNSET':
+                query = query.filter(
+                    or_(
+                        field == RetentionStatus.NOT_ELIGIBLE,
+                        and_(
+                            passive_criterion,
+                            or_(
+                                ParticipantSummary.ehrReceiptTime == None,
+                                ParticipantSummary.ehrReceiptTime <= eighteen_month_ago
+                            )
+                        )
+                    )
+                )
```

Comment:
I remember this slack conversation and this might line up with what I thought was right at the time, but looking back over everything I think Caroline's comment was a good clarification of everything: "A participant who is not eligible will always be not retained, but it is possible for someone who is eligible to also be not retained."

So she's saying that participants can be retention eligible, but not ACTIVE or PASSIVE. This is what I'm understanding now from the PRD:
- Participants are retention eligible if they meet some defined set of criteria (details not relevant here)
- Participants have an ACTIVE retention type if they are retention eligible and have done something something from a set of actions within the last 18 months
- Participants have a PASSIVE retention type if they are retention eligible and are not active, **but have EHR data in the last 18 months**

That last part of the definition of the PASSIVE type leaves another set of participants that _are_ retention eligible, don't have the ACTIVE retention type, and have not had EHR data in the last 18 months. That seems to fit with Caroline's comment and I believe it's that set of retention eligible participants that they're looking for with the UNSET retention type. (sorry for the really long comment)

Reply:
Just checked with Caroline, she wants to include no_eligible participants when filter retention type with `UNSET`.